pre_commit/languages/renv.py

```diff
@@ -0,0 +1,138 @@
+    with envcontext(get_env_patch(envdir)):
+        yield
+
+
+def get_env_dir(prefix: Prefix, version: str) -> str:
+    return prefix.path(helpers.environment_dir(ENVIRONMENT_DIR, version))
+
+
+def get_env_patch(venv: str) -> PatchesT:
+    return (
+        ('R_PROFILE_USER', os.path.join(venv, 'activate.R')),
+    )
+
+
+def install_environment(
+        prefix: Prefix,
+        version: str,
+        additional_dependencies: Sequence[str],
+) -> None:
+    env_dir = get_env_dir(prefix, version)
+    with clean_path_on_failure(env_dir):
+        os.makedirs(env_dir, exist_ok=True)
+        shutil.copy(prefix.path('DESCRIPTION'), env_dir)
+        shutil.copy(prefix.path('renv.lock'), env_dir)
+        cmd_output_b(
+            'Rscript', '--vanilla', '-e',
+            """
+            missing_pkgs <- setdiff(
+                "renv", unname(installed.packages()[, "Package"])
+            )
+            options(
+                repos = c(CRAN = "https://cloud.r-project.org"),
+                renv.consent = TRUE
+            )
+            install.packages(missing_pkgs)
+            renv::activate()
+            renv::restore()
+            activate_statement <- paste0(
+              'renv::activate("', file.path(getwd()), '"); '
+            )
+            writeLines(activate_statement, 'activate.R')
+            renv::install(normalizePath('.'))
+            """,
+            cwd=env_dir,
+        )
+        if additional_dependencies:
+            cmd_output_b(
+                'Rscript', '-e',
+                """
+                renv::install(commandArgs(trailingOnly = TRUE))
+                """, *additional_dependencies,
+                cwd=env_dir,
+            )
+
+
+def run_hook(
+        hook: Hook,
+        file_args: Sequence[str],
+        color: bool,
+) -> Tuple[int, bytes]:
+    with in_env(hook.prefix, hook.language_version):
+
+        return helpers.run_xargs(
+            hook, cmd_from_hook(hook), file_args, color=color,
+        )
+
+
+def cmd_from_hook(hook: Hook) -> Tuple[str, ...]:
+    opts = ('--no-save', '--no-restore', '--no-site-file', '--no-environ')
+    entry = shlex.split(hook.entry)
+    entry_validate(entry)
+
+    return (
+        *entry[:1], *opts,
+        *prefix_if_file_entry(entry, hook.prefix),
+        *hook.args,
+    )
+
+
+def entry_validate(entry: Sequence[str]) -> None:
+    """
+    # allowed entries:
+    # Rscript -e expr
+    # Rscript path/to/file
+    # TODO what if expr is x;2? not captured by shlex.split()
+    """
+    if entry[0] != 'Rscript':
+        raise SyntaxError('entry must start with `Rscript`.')
+
+    if entry[1] == '-e':
+        if len(entry) > 3:
+            raise SyntaxError('You can supply at most one expression.')
+    elif len(entry) > 2:
+        raise SyntaxError(
+            ' '.join([
+                'The only valid syntax is `Rscript -e {expr}`',
+                'or `Rscript path/to/hook/script`',
+            ]),
+        )
+
+    None
```

Comment:
this line does nothing and can be removed, note that python implicitly returns `None` for "void" functions and doesn't support implicit-return-without-`return`-keyword